pkg/controller/uniteddeployment/uniteddeployment_controller.go

```diff
@@ -434,26 +563,36 @@ func (r *ReconcileUnitedDeployment) updateStatus(instance *appsv1alpha1.UnitedDe
 	return err
 }
 
-func (r *ReconcileUnitedDeployment) calculateStatus(newStatus *appsv1alpha1.UnitedDeploymentStatus, nameToSubset *map[string]*Subset, nextReplicas, nextPartition *map[string]int32, currentRevision, updatedRevision *appsv1.ControllerRevision, control ControlInterface) *appsv1alpha1.UnitedDeploymentStatus {
+var extraStatusSelector = fmt.Sprintf(",%s=false", appsv1alpha1.ReservedPodLabelKey)
+
+func (r *ReconcileUnitedDeployment) calculateStatus(newStatus *appsv1alpha1.UnitedDeploymentStatus, existingSubsets *map[string]*Subset, nextReplicas, nextPartition *map[string]int32, currentRevision, updatedRevision *appsv1.ControllerRevision, control ControlInterface) *appsv1alpha1.UnitedDeploymentStatus {
 	expectedRevision := currentRevision.Name
 	if updatedRevision != nil {
 		expectedRevision = updatedRevision.Name
 	}
 
 	newStatus.Replicas = 0
+	newStatus.ReservedPods = 0
 	newStatus.ReadyReplicas = 0
 	newStatus.UpdatedReplicas = 0
 	newStatus.UpdatedReadyReplicas = 0
 
 	// sync from status
-	for _, subset := range *nameToSubset {
+	for name, subset := range *existingSubsets {
 		subsetReplicas, subsetReadyReplicas, subsetUpdatedReplicas, subsetUpdatedReadyReplicas := replicasStatusFn(subset)
-		newStatus.Replicas += subsetReplicas
+		newStatus.Replicas += subsetReplicas - subset.Status.UnschedulableStatus.UnavailablePods()
+		newStatus.ReservedPods += subset.Status.UnschedulableStatus.ReservedPods
 		newStatus.ReadyReplicas += subsetReadyReplicas
 		newStatus.UpdatedReplicas += subsetUpdatedReplicas
 		newStatus.UpdatedReadyReplicas += subsetUpdatedReadyReplicas
+		ss := newStatus.GetSubsetStatus(name)
+		ss.Replicas = subset.Status.Replicas
+		ss.ReadyReplicas = subset.Status.ReadyReplicas
+		ss.Partition = (*nextPartition)[name]
+		ss.ReservedPods = subset.Status.UnschedulableStatus.UnavailablePods()
 	}
```

Comment:
the func return not only the reserved pods, but also the PendingPods,  can we just return the reserved pods?